witness/golang/omniwitness/usbarmory/main.go

```diff
@@ -43,12 +45,58 @@ const (
 	// TODO(mhutchinson): these need to be read from file instead of constants
 	publicKey  = "TrustMe+68958214+AQ4Ys/PsXqfhPkNK7Y7RyYUMOJvfl65PzJOEiq9VFPjF"
 	signingKey = "PRIVATE+KEY+TrustMe+68958214+AZKby3TDZizdARF975ZyLJwGbHTivd+EqbfYTN5qr2cI"
+
+	// slotsPartitionOffsetBytes defines where our witness data storage partition starts.
+	// Changing this location is overwhelmingly likely to result in data loss.
+	slotsPartitionOffsetBytes = 1 << 30
+	// slotsPartitionLengthBytes specifies the size of the slots partition.
+	// Increasing this value is relatively safe, if you're sure there is no data
+	// stored in blocks which follow the current partition.
+	//
+	// We're starting with enough space for 10 slots of 1MB each.
+	slotsPartitionLengthBytes = 10 * slotSizeBytes
+
+	// slotSizeBytes is the size of each individual slot in the partition.
+	slotSizeBytes = 1 << 20
 )
 
+func init() {
+	debugConsole, _ := usbarmory.DetectDebugAccessory(250 * time.Millisecond)
+	<-debugConsole
+
+	if err := usbarmory.MMC.Detect(); err != nil {
+		glog.Exitf("Failed to detect MMC: %v", err)
+	}
+}
+
 func main() {
 	// We parse the flags despite declaring none ourselves so libraries are
 	// happy (looking at you, glog).
+	flag.Set("stderrthreshold", "INFO")
+	flag.Set("v", "1")
 	flag.Parse()
+
+	glog.Infof("Opening storage...")
+	part := openStorage()
+	glog.Infof("Storage opened.")
+
+	// Set this to true to "wipe" the storage.
+	// Currently this simply force-writes an entry with zero bytes to
+	// each known slot.
+	// If the journal(s) become corrupt a larger hammer will be required.
+	reinit := false
+	if reinit {
+		for i := uint(0); i < uint(part.NumSlots()); i++ {
+			s, err := part.Open(i)
+			if err != nil {
+				glog.Exitf("Failed to open slot %d: %v", i, err)
+			}
+			if err := s.Write([]byte{}); err != nil {
+				glog.Errorf("Failed to init slot %d: %v", i, err)
+			}
+		}
+	}
```

Comment:
Maybe move this to a method or something? It's a bit messy (and a little scary!) having a format operation right up here in the start of the main method.

What circumstances do you imagine this being called? It seems like maybe it's just while the storage is still under development? The other option is having this on a branch that isn't merged into master so we can always get it when we need it.

Reply:
Huh, my reply got lost :(

I've moved this to `Partition`, right now it's mainly just a development aid but it does seem reasonable for unikernel/users to want to be able to do some sort of `Factory Reset`.

I've also made it wipe all blocks allocated to slots, rather than just writing an empty record to all slots as it did previously - that seems more akin to the expected intent of an `Erase` call.

Reply:
Cool